src/NzbDrone.Core/Download/TrackedDownloads/DownloadMonitoringService.cs

```diff
@@ -113,12 +121,48 @@ private List<TrackedDownload> ProcessClientDownloads(IDownloadClient downloadCli
 
         private void RemoveCompletedDownloads(List<TrackedDownload> trackedDownloads)
         {
-            foreach (var trackedDownload in trackedDownloads.Where(c => c.DownloadItem.CanBeRemoved && c.State == TrackedDownloadStage.Imported))
+            var removableDownloads = trackedDownloads.Where(c =>
+            {
+                if (c.Protocol != DownloadProtocol.Torrent)
+                {
+                    return c.DownloadItem.CanBeRemoved && c.State == TrackedDownloadStage.Imported;
+                }
+
+                return TorrentHasFinishedSeeding(c);
+            });
+
+            foreach (var trackedDownload in removableDownloads)
             {
                 _eventAggregator.PublishEvent(new DownloadCompletedEvent(trackedDownload));
             }
         }
 
+        private bool TorrentHasFinishedSeeding(TrackedDownload download)
+        {
+            var historyItem = _historyService.Find(download.DownloadItem.DownloadId, HistoryEventType.Grabbed).Last();
+            if (historyItem == null)
+            {
+                return download.DownloadItem.CanBeRemoved && download.State == TrackedDownloadStage.Imported;
+            }
+
+            var indexer = _indexerFactory.All().First(i => i.Name == historyItem.Data["indexer"]);
```

Comment:
I think we should store the indexer ID, so a simple change in name doesn't break this.